source/js/components/petition/locales.js

```diff
@@ -0,0 +1,38 @@
+import data from './locale-strings.jsx';
+
+const DEFAULT_LOCALE = `en`;
+let currentLocale = false;
+
+/**
+ * This does a one-time fetch of the current locale based on the
+ * current URL. It simply checks what the very first part of the
+ * path-after-the-domain is, and tries to resolve that as a locale.
+ * @return {string} The current locale string
+ */
+function getCurrentLocale() {
+  if (typeof window === `undefined` || !window.location) {
+    return DEFAULT_LOCALE;
+  }
+
+  if (!currentLocale) {
+    var pathsegments = window.location.pathname.split(`/`).filter(v => v);
+
+    currentLocale = (pathsegments.length > 0) ? pathsegments[0] : DEFAULT_LOCALE;
+  }
+
+  return currentLocale || DEFAULT_LOCALE;
+}
+
+/**
+ * Get a localized string either for an explicitly known locale,
+ * or whatever is the current locale based on the page URL.
+ * @param {string} key the string key to localize
+ * @param {string} locale the locale for which to fetch the localized key (optional, defaults to current URL locale)
+ * @return {string} the localised key (with fall-back to the default localized string, and the key itself)
+ */
+function get(key, locale) {
+  locale = locale || getCurrentLocale();
+  return data[locale][key] || data[DEFAULT_LOCALE][key] || key;
```

Comment:
This doesn't quite seem to work? Non-localized locales aren't loading the CTA properly.
<img width="1552" alt="image" src="https://user-images.githubusercontent.com/1682681/44726093-f36a7980-aaa4-11e8-981e-3a30b58e5c55.png">